docs/reference/analysis/tokenizers.asciidoc

```diff
@@ -1,34 +1,135 @@
 [[analysis-tokenizers]]
 == Tokenizers
 
-Tokenizers are used to break a string down into a stream of terms
-or tokens. A simple tokenizer might split the string up into terms 
-wherever it encounters whitespace or punctuation.
+A _tokenizer_  receives a stream of characters, breaks it up into individual
+_tokens_ (usually individual words), and outputs a stream of _tokens_. For
+instance, a <<analysis-whitespace-tokenizer,`whitespace`>> tokenizer breaks
+text into tokens whenever it sees any whitespace.  It would convert the text
+`"Quick brown fox!"` into the terms `[Quick, brown, fox!]`.
 
-Elasticsearch has a number of built in tokenizers which can be
-used to build <<analysis-custom-analyzer,custom analyzers>>.
+The tokenizer is also responsible for recording the order or _position_ of
+each term (used for phrase and word proximity queries) and the start and end
+_character offsets_ of the original word which the term represents (used foor
+highlighting search snippets).
 
-include::tokenizers/standard-tokenizer.asciidoc[]
+Elasticsearch has a number of built in tokenizers which can be used to build
+<<analysis-custom-analyzer,custom analyzers>>.
 
-include::tokenizers/edgengram-tokenizer.asciidoc[]
+[float]
+=== Word Oriented Tokenizers
 
-include::tokenizers/keyword-tokenizer.asciidoc[]
+The following tokenizers are usually used for tokenizing full text into
+individual words:
+
+<<analysis-standard-tokenizer,Standard Tokenizer>>::
+
+The `standard` tokenizer divides text into terms on word boundaries, as
+defined by the Unicode Text Segmentation algorithm. It removes most
+punctuation symbols. It is the best choice for most languages.
+
+<<analysis-letter-tokenizer,Letter Tokenizer>>::
+
+The `letter` tokenizer divides text into terms whenever it encounters a
+character which is not a letter.
+
+<<analysis-letter-tokenizer,Lowercase Tokenizer>>::
+
+The `lowercase` tokenizer, like the `letter` tokenizer,  divides text into
```

Comment:
I'm kind of curious why we have this? It feels weird to have two ways to lowercase.